### ISSUE TITLE
[Feature Request] Option to only enforce dollar-variable-pattern at top-level scope

### ISSUE THREAD

**Issue by traviskaufman** [kristerkari/stylelint-scss#76]
We use SCSS-lint on [material design lite](https://github.com/google/material-design-lite/tree/master) and really like it. We use `scss/dollar-variable-pattern` to enforce consistency within our top-level sass API.

However, within functions, blocks, and mixins, we want to turn off that pattern, e.g.

``` scss
@mixin mdl-ripple-bg($config: ()) {
  /* stylelint-disable scss/dollar-variable-pattern */
  $config: map-merge(mdl-ripple-default-config(), $config);
  $pseudo: map-get($config, pseudo);
  /* stylelint-enable scss/dollar-variable-pattern */

  // ...
```

We've had to do this in a lot of places, and it would be great if there was an option that allowed enforcement of the pattern at the declaration-level only, rather than at the property level as well:

``` yaml
# .stylelintrc.yaml

scss/dollar-variable-pattern: 
  - ^_?mdl?-.+
  - topLevelOnly: true
```

Not sure if this is possible given the parser output being used to lint, but it would be a welcome addition 😄 also happy to help out on implementing it!

**Comment by dryoma** [comment]
Hi @traviskaufman, great to hear you make use of this plugin set. Sure, you can make a PR with this feature (or I will look into it a bit later). This rule is quite simple in implementation.

Following stylelint's naming conventions, it could be an option like 

```json
"ignore": "local"/"global",
```
(if I get it right, ignoring local variables is what you want - and then there really no reason to not allow for a reverse option value)

**Comment by kristerkari** [comment]
Looking at the example, I think that adding this option makes a lot of sense. 👍

**Comment by traviskaufman** [comment]
Awesome! Feel free to assign it to me and I'll tackle it this week.

**Comment by dryoma** [comment]
@traviskaufman The feature is available since 1.4.1

**Comment by traviskaufman** [comment]
Awesome!
